src/Forms/FieldsValidator.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace SilverStripe\Forms;
+
+/**
+ * Validates the internal state of all fields in the form.
+ */
+class FieldsValidator extends Validator
+{
+    public function php($data): bool
+    {
+        $valid = true;
+        $fields = $this->form->Fields();
+
+        foreach ($fields as $field) {
+            $valid = ($field->validate($this) && $valid);
```

Comment:
Can we break and return early on first `false` value? No need to iterate over 100 fields if the first one is invalid.

Reply:
At the same time not a big deal, I see we do exactly this in other validators so may as well keep it for consistency.

Reply:
We can, but then we won't get field messages added for all the invalid fields. Better to have messages for all invalid fields up front imo rather than fixing one invalid value only for another to fail validation immediately afterwards.

Reply:
Yeah, I forgot that this has a side effect of adding the messages while validating.